### PR TITLE
Remove jenkins section, add config_sort reminder in README

### PR DESCRIPTION
**Pull request by krzyzacy**
also give @k8s-ci-robot a link

/area documentation
/assign @BenTheElder

### CODE REVIEW

**Comment by krzyzacy** [comment]
```
W0302 19:13:41.570] Run: ('./hack/verify-bazel.sh',)
W0302 19:13:49.411] ../../internal/config/directives.go:24:2: cannot find package "github.com/bazelbuild/buildtools/build" in any of:
W0302 19:13:49.412] 	/tmp/tmp.rB3ZYqljQr/src/github.com/bazelbuild/bazel-gazelle/vendor/github.com/bazelbuild/buildtools/build (vendor tree)
W0302 19:13:49.412] 	/usr/local/go/src/github.com/bazelbuild/buildtools/build (from $GOROOT)
W0302 19:13:49.412] 	/tmp/tmp.rB3ZYqljQr/src/github.com/bazelbuild/buildtools/build (from $GOPATH)
W0302 19:13:49.413] ../../internal/rules/construct.go:27:2: cannot find package "github.com/bazelbuild/buildtools/tables" in any of:
W0302 19:13:49.413] 	/tmp/tmp.rB3ZYqljQr/src/github.com/bazelbuild/bazel-gazelle/vendor/github.com/bazelbuild/buildtools/tables (vendor tree)
W0302 19:13:49.413] 	/usr/local/go/src/github.com/bazelbuild/buildtools/tables (from $GOROOT)
W0302 19:13:49.414] 	/tmp/tmp.rB3ZYqljQr/src/github.com/bazelbuild/buildtools/tables (from $GOPATH)
```

/shrug
cc @BenTheElder @ixdy

**Comment by ixdy** [comment]
hmm, I wonder if kubernetes/kubernetes#60442 is back...

File: README.md [hunk 1]
```diff
@@ -108,7 +110,7 @@ We are moving towards making more jobs to fit into the generated config.
 
 Presubmit will tell you if you forget to do any of this correctly.
 
-Merge your PR and @k8s-ci-robot will deploy your change automatically.
+Merge your PR and [@k8s-ci-robot](https://github.com/k8s-ci-robot) will deploy your change automatically.
```
Comment (ixdy) [inline README.md]: it still refuses to create a link?

Comment (krzyzacy) [reply README.md]: yeah, at least current https://github.com/kubernetes/test-infra does not have a link

File: README.md [hunk 2]
```diff
@@ -64,10 +64,12 @@ you'll need to do the following:
...
-* (Deprecated!) Some old jobs still run on jenkins
-  - Please do not add new jobs to jenkins
-  - Jenkins configuration is defined at `jenkins/job-configs`
-  - More deprecated details at [jenkins/README.md](jenkins/README.md)
+
+The configs need to be sorted, or else presubmit will fail.
+To keep them sorted, you can run:
+```
+bazel run //jobs:config_sort
+```
```
Comment (BenTheElder) [inline README.md]: maybe just mention `hack/update-config.sh`? /shrug

Comment (BenTheElder) [reply README.md]: They need to be sorted *and* kubernetes must be in sync with the security repo

Comment (krzyzacy) [reply README.md]: done

**Comment by krzyzacy** [comment]
PTAL?

File: README.md (hunk 1 above)
Comment (ixdy) [reply README.md]: oh, I was looking at the wrong section in https://github.com/krzyzacy/test-infra/blob/ba4ac00d79243f52359ddcaa0d905f11cf89cc7e/README.md. 

there's another @k8s-ci-robot reference on L133 that may want to be updated, too.

Comment (krzyzacy) [reply README.md]: done

**Pull request merged**